### PR TITLE
[Tests] Fix test_docker_storage_mounts for azure blob

### LINKED ISSUE #3781: [Azure] `test_docker_storage_mounts` needs to be updated for azure blob storage
`pytest tests/test_smoke.py::test_docker_storage_mounts --azure` fails with:

```
+ aws s3 ls sky-test-17217151744714668/hello.txt || gsutil ls gs://sky-test-17217151744714668/hello.txt

An error occurred (NoSuchBucket) when calling the ListObjectsV2 operation: The specified bucket does not exist
BucketNotFoundException: 404 gs://sky-test-17217151744714668 bucket does not exist.
``` 

We probably need to update this test to check azure blob storage as well. 

Blocking 0.6.1.

### PR DESCRIPTION
**Pull request by romilbhardwaj**
Closes #3781.

Tested: `pytest tests/test_smoke.py::test_docker_storage_mounts --azure`

### CODE REVIEW

File: tests/test_smoke.py [hunk 1]
```diff
@@ -1189,9 +1190,17 @@ def test_docker_storage_mounts(generic_cloud: str, image_id: str):
...
     # ubuntu 18.04 does not support fuse3, and blobfuse2 depends on fuse3.
     azure_mount_unsupported_ubuntu_version = '18.04'
+    # Commands to verify bucket upload. We need to check all three
+    # storage types because the optimizer may pick any of them.
+    s3_command = f'aws s3 ls {storage_name}/hello.txt'
+    gsutil_command = f'gsutil ls gs://{storage_name}/hello.txt'
+    azure_blob_command = TestStorageWithCredentials.cli_ls_cmd(
+        storage_lib.StoreType.AZURE, storage_name, suffix='hello.txt')
     if azure_mount_unsupported_ubuntu_version in image_id:
+        include_private_mount = False if generic_cloud == 'azure' else True
```
Comment (Michaelvll) [inline tests/test_smoke.py:1200]: can we add a comment for why private bucket is not allowed for azure?

Comment (romilbhardwaj) [reply tests/test_smoke.py:1205]: Added!

Comment (Michaelvll) [reply tests/test_smoke.py:1205]: Let's file an issue for this to check the dependencies earlier in the `setup.py`?

**Review by Michaelvll — APPROVED** [review]
Thanks for the fix @romilbhardwaj! LGTM.

**Pull request merged**